curvlinops/ggn.py

```diff
@@ -13,44 +13,53 @@
 
 
 def make_ggn_vector_product(
-    f: Callable[..., Tensor], c: Callable[[Tensor, Tensor], Tensor]
-) -> Callable[
-    [Tuple[Tensor, ...], Tensor, Tensor, Tuple[Tensor, ...]], Tuple[Tensor, ...]
-]:
+    f: Callable[..., Tensor], c: Callable[..., Tensor], c_extra_args: int = 0
+) -> Callable[..., Tuple[Tensor, ...]]:
     """Create a function that computes GGN-vector products for given f and c functions.
 
     Args:
         f: Function that takes parameters and input, returns prediction.
             Signature: (*params, X) -> prediction
-        c: Function that takes prediction and target, returns loss.
-            Signature: (prediction, y) -> loss
+        c: Function that takes prediction, target, and optional additional args.
+            Signature: (prediction, y, *args) -> loss
+        c_extra_args: Number of additional arguments that the loss function c expects
```

Comment:
Nit: Based on the name I would expect the actual extra arguments for `c`, could be clarified by changing to something like `num_c_extra_args`.